db/migrate/20200519141015_create_storage_resources.rb

```diff
@@ -0,0 +1,16 @@
+class CreateStorageResources < ActiveRecord::Migration[5.1]
+  def change
+    create_table :storage_resources do |t|
+      t.string :name
+      t.bigint :ems_id
```

Comment:
I'm concerned that these don't have indexes on any of these foreign keys, `:references` is a nice way to do this in one shot e.g. `t.references :ems, :type => :bigint, :index => true, :references => :ext_management_system`

Reply:
@agrare , t.references is a front to https://apidock.com/rails/v6.0.0/ActiveRecord/ConnectionAdapters/SchemaStatements/add_reference? it's a bit odd, because the documentation seems to say that it's `:ref_name`, not `:references` ? Is the documentation incorrect? Or am I looking in the wrong place?

While I'm doing the changes, could you explain why not use `:foreign_key => true`  instead of `:index => true`? Why don't we want DB-level foreign keys? Because we might want to convert the relations to polymorphic?

Also, doesn't :type default to bignit?

Reply:
>  it's a bit odd, because the documentation seems to say that it's :ref_name, not :references ? Is the documentation incorrect? Or am I looking in the wrong place?

No that's correct but `ref_name` is just the name of the parameter, `add_reference(table_name, ref_name, **options) public` so in this case `:ems` is the `ref_name` parameter and the rest of the options hash goes to the `options` parameter

Reply:
> Also, doesn't :type default to bignit?

In rails 5.2 it does, but prior to that it defaulted to `:integer`.  Take a look at https://apidock.com/rails/v5.2.3/ActiveRecord/ConnectionAdapters/SchemaStatements/add_reference versus https://apidock.com/rails/v5.1.7/ActiveRecord/ConnectionAdapters/SchemaStatements/add_reference and it looks like you generated this on 5.1 so it would have defaulted to `:integer`